platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,346 @@
+    * EC Key generation will have to happen before a node can join the network.
+* A new node added to an existing network will need to be given a state from an existing node after the network has
+  adopted the consensus roster containing the new node.
+
+Implications Of Completion
+
+* The consensus nodes will be able to create ledger signatures for use in block proofs.
+
+### Requirements
+
+TSS Core Requirements
+
+| Requirement ID | Requirement Description                                                                                                      |
+|----------------|------------------------------------------------------------------------------------------------------------------------------|
+| TSS-001        | The ledger MUST have a public/private signing key pair that does not change when the consensus roster changes.               |
+| TSS-002        | The ledger private key MUST be a secret that nobody knows.                                                                   |
+| TSS-003        | The `ledger id` MUST be a public key that everyone knows and able to verify signatures from the ledger private key.          |
+| TSS-004        | The ledger signature for a message MUST be produced through a threshold number of consensus nodes signing the message.       |
+| TSS-005        | There MUST be a way to bootstrap an existing network's consensus roster with to give the ledger a public/private signing key |
+| TSS-006        | There MUST be a way for a new network to generate a public/private signing key for the ledger.                               |
+| TSS-007        | There MUST be a way to rotate the ledger's public/private signing key.                                                       |
+| TSS-008        | The ledger signature SHOULD be verifiable by an EVM smart contract without excessive cost or time.                           |
+| TSS-009        | The TSS implementation SHOULD be able to switch elliptic curves.                                                             |
+| TSS-010        | The TSS algorithm SHOULD be able to model consensus weight with high precision.                                              |
+
+### Design Decisions
+
+#### New Address Book Life-Cycle
+
+This design proposal will deliver after Dynamic Address Book  (DAB) Phase 2 and before DAB Phase 3. In Phase 2 the
+address book is updated on software upgrades. In Phase 3 the address book is updated dynamically without restarting
+the node. Since the TSS effort requires keying a roster with enough `TssMessages` to generate ledger signatures, a
+modified life-cycle is needed. Some of the dynamic work in Phase 3 will be needed. This work is being encapsulated
+in the TSS-Roster proposal, which will need to be completed before the TSS-Ledger-Id proposal can be fully implemented.
+
+The relevant Address Book life-cycle changes are the following:
```

Comment:
After reading it 2 times, I still do not understand the life-cycle. The description is good; I assume it is just too complex. Would it be possible to add a graph/diagram?

Reply:
I am adding a diagram.   The `TSS-Roster` proposal will outline the structure of the roster.